server/src/main/java/fr/uga/l3miage/tp4/models/UserEntity.java

```diff
@@ -0,0 +1,20 @@
+package fr.uga.l3miage.tp4.models;
+
+import javax.persistence.*;
+
+@MappedSuperclass
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+public abstract class UserEntity {
+
+    @Id
+    private Long id;
+
+    private String firstname;
+    private String lastname;
+    @Column(nullable = false, unique = true)
+    private String email;
+    @Column(unique = true)
+    private String phoneNumber;
```

Comment:
c'était dans le sujet 
```suggestion
    @Column(unique = true, length = 8)
    private String phoneNumber;
```